src/Dialect/ONNX/Rewrite.cpp

```diff
@@ -557,36 +566,146 @@ void ONNXAddOp::getCanonicalizationPatterns(
   results.insert<FuseAddConvNullBiasPattern>(context);
 }
 
+/// on the ONNXCastOp.
+void ONNXCastOp::getCanonicalizationPatterns(
+    RewritePatternSet &result, MLIRContext *context) {
+  result.insert<CastEliminationPattern>(context);
+  result.insert<FuseCastCastPattern>(context);
```

Comment:
This pattern causes a loss of clipping behaviour if you do f32->i8->f32.

Mind if we add a constraint to ensure the clipping behavior is preserved? Or alternative solutinos? 

For vis: @philass @sorenlassen

Reply:
I merged PR #2211 to disable the pattern. Issue #2210 tracks the effort to replace it with a version that fuses only casts that are sound to fuse.